source/client_channel_handler.c

```diff
@@ -797,12 +803,25 @@ uint16_t mqtt_create_request(
         mqtt_connection_unlock_synced_data(connection);
     } /* END CRITICAL SECTION */
     if (should_schedule_task) {
-        AWS_LOGF_TRACE(
-            AWS_LS_MQTT_CLIENT,
-            "id=%p: Currently not in the event-loop thread, scheduling a task to send message id %" PRIu16 ".",
-            (void *)connection,
-            next_request->packet_id);
-        aws_channel_schedule_task_now(connection->slot->channel, &next_request->outgoing_task);
+        /* It's possible that after we release the lock, the slot is gone. So, let's check the slot. */
+        if (connection->slot && connection->slot->channel) {
+            AWS_LOGF_TRACE(
+                AWS_LS_MQTT_CLIENT,
+                "id=%p: Currently not in the event-loop thread, scheduling a task to send message id %" PRIu16 ".",
+                (void *)connection,
+                next_request->packet_id);
+            aws_channel_schedule_task_now(connection->slot->channel, &next_request->outgoing_task);
```

Comment:
it probably will not solve the root cause... but we cannot put a lock around the `aws_channel_schedule_task_now` function...

Reply:
use aws_channel_acquire_hold() to keep it alive until you run this (don't forget to release_hold() after ;) )